src/workers/newView.ts

```diff
@@ -88,37 +55,49 @@ const incrementReadingStreak = async (
         currentStreak: 1,
         totalStreak: 1,
         maxStreak: 1,
-        lastViewAt: viewTime,
+        lastViewAt: timestamp,
       }),
     );
-  } else {
-    const results = await repo.query(INC_STREAK_QUERY, [
-      userId,
-      viewTime,
-      DEFAULT_TIMEZONE,
-    ]);
 
-    const { currentStreak, didIncrementStreak } = results?.[0] ?? {};
+    return true;
+  }
 
-    if (didIncrementStreak) {
-      // milestones are currently defined on fibonacci sequence
-      const showStreakMilestone = isFibonacci(currentStreak);
-      await con.getRepository(Alerts).save({
-        userId,
-        showStreakMilestone,
-      });
-    }
+  if (
+    streak &&
+    (!streak.lastViewAt ||
+      streak.currentStreak === 0 ||
+      streak.lastViewAt.getDate() !== timestamp.getDate())
```

Comment:
Both `lastViewAt` and `timestamp` are in UTC by default. This code does not deal with user timezones at all and will increment the reading streak only with UTC in mind, not the user timezone. I'm not sure that is the intention.

Imagine the following scenarios:
# User in PDT
1. User is in PDT (UTC -08:00)
2. User reads an article at 12:00:00 local time, which is 20:00 UTC. We increment the streak
3. User reads article at 19:00:00 local time, same date, which is 3AM UTC the next day. We end up incrementing the streak, although we should not have.

# User in Sydney
1. User is in Sydney (UTC +10:00)
2. User reads an article at 16:00 local time, which is 2AM UTC. We increment the streak.
3. User reads an article the next day at 7:00 local time, which is 17:00 UTC, the same day as was the 2AM UTC view. We do not increment the streak, even if should have done it.

Does that make sense?